tratto/src/main/java/star/tratto/util/javaparser/TypeUtils.java

```diff
@@ -0,0 +1,295 @@
+package star.tratto.util.javaparser;
+
+import com.github.javaparser.TokenRange;
+import com.github.javaparser.ast.body.CallableDeclaration;
+import com.github.javaparser.ast.body.Parameter;
+import com.github.javaparser.ast.body.TypeDeclaration;
+import com.github.javaparser.ast.type.TypeParameter;
+import org.plumelib.reflection.Signatures;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * This class provides static methods to convert between the ClassGetName and
+ * ClassGetSimpleName forms of Java types, as well as other utilities for
+ * analyzing various type representations.
+ */
+public class TypeUtils {
+    /**
+     * All primitive field descriptors. The ClassGetName form of a type uses
+     * field descriptor for arrays of primitive types.
+     */
+    private final static List<String> primitiveFieldDescriptors = List.of("Z", "B", "C", "S", "I", "J", "F", "D");
+
+    /** Private constructor to avoid creating an instance of this class. */
+    private TypeUtils() {
+        throw new UnsupportedOperationException("This class cannot be instantiated.");
+    }
+
+    /**
+     * Gets the package name from a ClassGetName form of a type. If the
+     * package does not exist (e.g. default or primitive), then the method
+     * returns an empty string.
+     *
+     * @param classGetName a ClassGetName form of a type
+     * @return the package name of the class
+     */
+    public static String getPackageNameFromClassGetName(String classGetName) {
+        List<String> nameSegments = Arrays.asList(classGetName.split("\\."));
+        List<String> packageSegments = nameSegments.subList(0, nameSegments.size() - 1);
+        return String.join(".", packageSegments);
+    }
+
+    /**
+     * Gets the innermost class from a ClassGetName form of a type.
+     *
+     * @param classGetName a ClassGetName form of a type
+     * @return the innermost class of the type
+     */
+    public static String getInnermostClassNameFromClassGetName(String classGetName) {
+        List<String> nameSegments = Arrays.asList(classGetName.split("[.$]"));
```

Comment:
This implementation is unnecessarily efficient.  Instead, use `lastIndexOf` and `substring`.